custom_metrics/origin-trials.js

```diff
@@ -0,0 +1,156 @@
+//[origin_trials]
+/**
+ * Origin Trials
+ * https://github.com/GoogleChrome/OriginTrials
+ *
+ * Test site: https://floc-ot-meta.glitch.me/
+ * Test site: https://floc-ot-header.glitch.me/
+ */
+
+/**
+ * https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/check-token.html
+ *
+ * @function validate
+ * @param {string} tokenElem
+ * @returns {object} origin_trial_metadata
+ */
+function validate(tokenElem) {
+  let validityElem,
+    versionElem,
+    originElem,
+    subdomainElem,
+    thirdpartyElem,
+    usageElem,
+    featureElem,
+    expiryElem;
+
+  const utf8Decoder = new TextDecoder('utf-8', {fatal: true});
+
+  // Base64-decode the token into a Uint8Array.
+  let tokenStr;
+  try {
+    tokenStr = atob(tokenElem);
+  } catch (e) {
+    console.error(e);
+    validityElem = 'Invalid Base64';
```

Comment:
What's the purpose of assigning to `validityElem` before returning? Doesn't seem like it's ever used.

Maybe you should initialize `origin_trial_metadata` earlier, update the validity property as needed, and return the entire object when there are processing errors. That way it'll still include the validity data but none of the other processed data.